okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java

```diff
@@ -537,14 +550,32 @@ static SslSocketFactoryResult sslSocketFactoryFrom(ChannelCredentials creds) {
         return SslSocketFactoryResult.error(
             "TLS features not understood: " + incomprehensible);
       }
-      SSLSocketFactory sslSocketFactory;
+      KeyManager[] km = null;
+      if (tlsCreds.getKeyManagers() != null) {
+        km = tlsCreds.getKeyManagers().toArray(new KeyManager[0]);
+      } else if (tlsCreds.getPrivateKey() != null) {
+        return SslSocketFactoryResult.error("byte[]-based private key unsupported. Use KeyManager");
```

Comment:
In a previous comment you said "Users tend to have bytes and not Certificate or PrivateKey. ...". We don't actually support that use case here. BTW `io.grpc.xds.internal.sds.trust.CertificateUtils.getPrivateKey` converts a byte array to a `PrivateKey` and it doesn't depend on netty...